src/vos/vos_obj.c

```diff
@@ -459,9 +459,6 @@ vos_obj_delete(daos_handle_t coh, daos_unit_oid_t oid)
 	rc = umem_tx_end(umm, rc);
 	if (rc)
 		goto out;
-
-	/* NB: noop for full-stack mode */
-	gc_wait();
```

Comment:
Why it's removed?

Reply:
For testing purposes mainly.  It's only for unit tests and I wanted the gc tests to actually test garbage collection after the fact and not having that gives me more control over when it runs.   It really should have no effect on anything but unit tests since only reintegration uses these APIs and it's a noop in that case.